tests/Agent/IntegrationTests/SharedApplications/Common/MultiFunctionApplicationHelpers/NetStandardLibraries/AwsSdk/InvokeLambdaExerciser.cs

```diff
@@ -0,0 +1,96 @@
+// Copyright 2020 New Relic, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.IO;
+using System.Security.AccessControl;
+using System.Threading.Tasks;
+using Amazon;
+using NewRelic.Agent.IntegrationTests.Shared.ReflectionHelpers;
+using NewRelic.Api.Agent;
+
+namespace MultiFunctionApplicationHelpers.NetStandardLibraries.AwsSdk
+{
+    [Library]
+    public class InvokeLambdaExerciser
+    {
+        [LibraryMethod]
+        [Transaction]
+        public void InvokeLambdaSync(string function, string payload)
+        {
+#if NETFRAMEWORK
+            var client = new Amazon.Lambda.AmazonLambdaClient(RegionEndpoint.USWest2);
+            var request = new Amazon.Lambda.Model.InvokeRequest
+            {
+                FunctionName = function,
+                Payload = payload
+            };
+
+            // Note that we aren't invoking a lambda that exists! This is guaranteed to fail, but all we care
+            // about is that the agent is able to instrument the call.
+            try
+            {
+                var response = client.Invoke(request);
+            }
+            catch
```

Comment:
Should we only catch a specific failure here to help with detecting cases where instrumentation results in an exception that isn't caught?

Reply:
We can try it. Since we aren't sending a valid request I'm not sure we're guaranteed to get a consistent error back, but local testing seems to work.

Reply:
It looks like we're seeing a 400 Bad Request error from the CI pipeline, instead of the ResourceNotFound error. Your suspicions about the error not being consistent seem to be valid. Maybe, there's a way to check that the error isn't coming from the agent?

Reply:
I was hoping it would at least throw something in the `Amazon` namespace so we could check the type, but no such luck. Another option would be to call a real Lambda but that felt unnecessary.

Reply:
I wrote a helper utility to check the call stack to see if the Exception came from our agent or the integration test app, and it worked, but proved not especially useful. If the Agent throws, we'll miss the Harvest, and everything in the test will fail anyway, and the Agent log will show the Exception.